relayer/src/util/task.rs

```diff
@@ -90,14 +90,19 @@ pub fn spawn_background_task<E: Display>(
     interval_pause: Option<Duration>,
     mut step_runner: impl FnMut() -> Result<Next, TaskError<E>> + Send + Sync + 'static,
 ) -> TaskHandle {
-    info!("spawning new background task {}", task_name);
+    let span = span!(tracing::Level::ERROR, "task", name = %task_name);
+    let _entered = span.enter();
+
+    info!("spawning");
```

Comment:
Cool! 

I noticed that we can now get into a 3-level span, e.g., `task{..}:task{..}:relay{..}`.


Sample log output:
```
2022-01-11T08:36:32.047359Z  INFO ThreadId(01) Hermes has started
2022-01-11T08:36:47.169785Z  INFO ThreadId(2847) task{name=supervisor_batch}:task{name=PacketCmdWorker(ibc-0:transfer/channel-0 -> ibc-1)}: spawning
2022-01-11T08:36:47.170052Z  INFO ThreadId(2847) task{name=supervisor_batch}:task{name=PacketWorker(ibc-0:transfer/channel-0 -> ibc-1)}: spawning
2022-01-11T08:36:47.433581Z  INFO ThreadId(2851) task{name=supervisor_batch}:task{name=PacketCmdWorker(ibc-0:transfer/channel-0 -> ibc-1)}:relay{odata=BF24AQfUzl ->Destination @0-18454; len=1}: [ibc-0:transfer/channel-0 -> ibc-1] assembled batch of 2 message(s)
2022-01-11T08:36:47.436739Z  INFO ThreadId(18) [ibc-1] refresh: retrieved account sequence=109 number=1
2022-01-11T08:36:47.448264Z  INFO ThreadId(2851) task{name=supervisor_batch}:task{name=PacketCmdWorker(ibc-0:transfer/channel-0 -> ibc-1)}:relay{odata=BF24AQfUzl ->Destination @0-18454; len=1}: [Async~>ibc-1] response(s): 1; Ok:78DD8796BA5C98F34EE531F6156BE608A3456A3CE2F7EE2C52B05030BAA86A0A

2022-01-11T08:36:47.448364Z  INFO ThreadId(2851) task{name=supervisor_batch}:task{name=PacketCmdWorker(ibc-0:transfer/channel-0 -> ibc-1)}:relay{odata=BF24AQfUzl ->Destination @0-18454; len=1}: [ibc-0:transfer/channel-0 -> ibc-1] success
2022-01-11T08:36:55.751340Z  INFO ThreadId(2847) task{name=supervisor_batch}:task{name=PacketCmdWorker(ibc-1:transfer/channel-0 -> ibc-0)}: spawning
2022-01-11T08:36:55.751602Z  INFO ThreadId(2847) task{name=supervisor_batch}:task{name=PacketWorker(ibc-1:transfer/channel-0 -> ibc-0)}: spawning
2022-01-11T08:36:56.010186Z  INFO ThreadId(2871) task{name=supervisor_batch}:task{name=PacketCmdWorker(ibc-1:transfer/channel-0 -> ibc-0)}:relay{odata=LNDPziRenQ ->Destination @1-18452; len=1}: [ibc-1:transfer/channel-0 -> ibc-0] assembled batch of 2 message(s)
2022-01-11T08:36:56.013194Z  INFO ThreadId(17) [ibc-0] refresh: retrieved account sequence=135 number=1
2022-01-11T08:36:56.024201Z  INFO ThreadId(2871) task{name=supervisor_batch}:task{name=PacketCmdWorker(ibc-1:transfer/channel-0 -> ibc-0)}:relay{odata=LNDPziRenQ ->Destination @1-18452; len=1}: [Async~>ibc-0] response(s): 1; Ok:FFD0C8779F0C188A0828CF5FEE9E29A336E6BE60EEA76A8075FAA80717F3D6EF

2022-01-11T08:36:56.024290Z  INFO ThreadId(2871) task{name=supervisor_batch}:task{name=PacketCmdWorker(ibc-1:transfer/channel-0 -> ibc-0)}:relay{odata=LNDPziRenQ ->Destination @1-18452; len=1}: [ibc-1:transfer/channel-0 -> ibc-0] success
```

Feedback:

I think this improves debugging, as the relations among task names becomes explicit in each log line.

As a disadvantage: this can make log lines very long. So let's improve that.
- In particular the span `task{name=supervisor_batch}` not sure how it improves observability, so maybe we can skip entering that span.
- The span `task{name=PacketCmdWorker(ibc-0:transfer/channel-0 -> ibc-1)}` seems useful, and we should probably keep it. We have redundant information now, however, since the path details `[ibc-0:transfer/channel-0 -> ibc-1]` appears twice on most lines, see eg. lines with "assembled batch of 2 message(s)". This information appears first from the `span` we just introduced, and second from `RelayPath:fmt::Display` impl ([example](https://github.com/informalsystems/ibc-rs/blob/85c60fa8f1b37e0f823aaa3cf03c79bafd8ef994/relayer/src/link/operational_data.rs#L179)). I think we should eliminate the second and keep the span only.

Reply:
Thanks, I haven't noticed there is already a "task" span. I can revert the top-level messages in `spawn_background_task` to use the `bg_task` keyword with the background task name.

Reply:
Oh, this is more interesting; there are two layers of background tasks, one spawning another. In principle, this indirect recursion can go arbitrarily deep. So maybe a span is not a good solution.

Reply:
> * This information appears first from the `span` we just introduced, and second from `RelayPath:fmt::Display` impl ([example](https://github.com/informalsystems/ibc-rs/blob/85c60fa8f1b37e0f823aaa3cf03c79bafd8ef994/relayer/src/link/operational_data.rs#L179)).

Yes, there are many `[{}]` constructs in the log macro invocations, using a `Display` impl of this or that object to provide context information. I think they should all be converted to use spans instead.

Reply:
I have reworked the span use in `spawn_background_task`, it should look much better now.

Reply:
> * We have redundant information now, however, since the path details `[ibc-0:transfer/channel-0 -> ibc-1]` appears twice on most lines, see eg. lines with "assembled batch of 2 message(s)". This information appears first from the `span` we just introduced, and second from `RelayPath:fmt::Display` impl ([example](https://github.com/informalsystems/ibc-rs/blob/85c60fa8f1b37e0f823aaa3cf03c79bafd8ef994/relayer/src/link/operational_data.rs#L179)). I think we should eliminate the second and keep the span only.

I have removed the `Display` impl for `RelayPath` and converted its uses into the same information laid out with spans.